resources/idlharness.js

```diff
@@ -218,6 +226,81 @@ IdlArray.prototype.add_untested_idls = function(raw_idls, options)
     this.internal_add_idls(parsed_idls, options);
 };
 
+//@}
+IdlArray.prototype.add_dependency_idls = function(raw_idls, options)
+//@{
+{
+    const parsed_idls = WebIDL2.parse(raw_idls);
+    const new_options = { only: [] }
+    
+    const all_deps = new Set();
+    Object.values(this["inheritance"]).forEach(v => all_deps.add(v));
+    Object.values(this["implements"]).forEach(v => all_deps.add(v));
+    this.partials.map(p => p.name).forEach(v => all_deps.add(v));
+    
+    console.log('all deps', all_deps);
+
+    if (options && options.except && options.only) {
+        throw new IdlHarnessError("The only and except options can't be used together.");
+    }
+    const skip = name => {
```

Comment:
This is very similar to the `function should_skip(name)` in `internal_add_idls`. Can you copy that so that any differences stand out, or, if it seems worthwhile, use a shared definition?